client/src/components/App.tsx

```diff
@@ -47,13 +51,39 @@ const App = (): ReactElement => {
 
   const location = useLocation();
 
+
+  const [mode, setMode] = React.useState<'light' | 'dark'>('light');
+  const colorMode = React.useMemo(
+    () => ({
+      toggleColorMode: () => {
+        setMode((prevMode) => (prevMode === 'light' ? 'dark' : 'light'));
+      },
+    }),
+    [],
+  );
+
+  const theme = React.useMemo(
+    () =>
+      createTheme({
+        palette: {
+          mode,
+        },
+      }),
+    [mode],
+  );
+
```

Comment:
Lines 54-74 I want to factor out of App.tsx later so that App.tsx stays simple and clean. As of now this is key for MVP functionality.

Reply:
We can look at refactoring into a context object that can hold all of these values so we have a single context provider. Not sure if that would be more efficient  but something to consider.